common/scala/src/main/scala/org/apache/openwhisk/core/entity/WhiskAction.scala

```diff
@@ -351,6 +361,49 @@ object WhiskAction extends DocumentFactory[WhiskAction] with WhiskEntityQueries[
   val execFieldName = "exec"
   val requireWhiskAuthHeader = "x-require-whisk-auth"
 
+  // annotation permission key name
+  val permissionsFieldName = "permissions"
```

Comment:
I am not quite sure it is a good idea to use annotations for permission management.
I feel like we need a new protocol.

Reply:
yes, i also feel it is not a good idea to use annotation for permission management.
Currently, have no better idea for it.

Reply:
I think we can store permission information in DB and fetch it along with `Identity` information.
Some proper tools to manage permission are required and the same change should be applied in CLI(`wsk` or at least `wskadmin`) as well.

Reply:
I think such permission information is applied to actions instead of `Identity`

Reply:
I meant, we can fetch the permission information in a similar way with `Identity`. 
This information can be stored in the cache too.
Permission needs to be checked before actual operations.